README
Alldeb installer
================

Alldeb installer merupakan program untuk melakukan instalasi paket-paket
Ubuntu DEB yang dikemas lagi menjadi file ALLDEB. File ini dikumpulkan 
berdasarkan keterkaitan atau dependensi dari satu meta-paket.

File alldeb dibuat dengan alldeb_maker.sh yang disertakan bersama dengan 
kode sumber ini.

Kompilasi
---------

Untuk mengompilasi kode sumber ini, anda memerlukan paket-paket untuk 
building, antara lain: cmake, build-essential, libqt4-dev, qt4-qmake.

Perintah standar untuk building adalah:

     mkdir build
     cmake ..
     make
     sudo make install
     
Skrip cmake belum menyertakan aturan untuk uninstall, maka untuk mencopot
instalasi harus dilakukan secara manual.

     sudo rm /usr/bin/alldeb-installer
     sudo rm -rf /usr/share/alldeb
     


debian/changelog
alldeb-installer (0.2.1-ubuntu1) trusty; urgency=medium

  * New minor version commit

 -- Slamet Badwi (bedouin) <slamet.badwi@gmail.com>  Sat, 24 Jan 2015 18:03:41 +0700

alldeb-installer (0.2-ubuntu10) trusty; urgency=medium

  * New logic for checking PolicyKit agent presence (Closes: #1406335)

 -- Slamet Badwi (bedouin) <slamet.badwi@gmail.com>  Sat, 24 Jan 2015 07:42:50 +0700

alldeb-installer (0.2-ubuntu9) trusty; urgency=medium

  * Added Policykit agent programs as dependency

 -- Slamet Badwi (bedouin) <slamet.badwi@gmail.com>  Tue, 30 Dec 2014 13:51:48 +0700

alldeb-installer (0.2-ubuntu8) trusty; urgency=medium

  * Manually merged alldeb-installer next branch, once more.

 -- Slamet Badwi (bedouin) <slamet.badwi@gmail.com>  Sun, 28 Dec 2014 11:31:29 +0700

alldeb-installer (0.2-ubuntu7) trusty; urgency=low

  * Merged with alldeb-installer next branch.

 -- Slamet Badwi (bedouin) <slamet.badwi@gmail.com>  Mon, 12 May 2014 18:55:23 +0700

alldeb-installer (0.2-ubuntu6) trusty; urgency=low

  * Edited apt-ftparchive command

 -- Slamet Badwi (bedouin) <slamet.badwi@gmail.com>  Mon, 05 May 2014 13:22:26 +0700

alldeb-installer (0.2-ubuntu5) trusty; urgency=low

  * Modified source, rearranged objects

 -- Slamet Badwi (bedouin) <slamet.badwi@gmail.com>  Fri, 25 Apr 2014 19:35:13 +0700

alldeb-installer (0.2-ubuntu1) trusty; urgency=low

  * First beta version

 -- Slamet Badwi (bedouin) <slamet.badwi@gmail.com>  Thu, 24 Apr 2014 21:16:00 +0700

alldeb-installer (0.1-ubuntu6) trusty; urgency=low

  * Redefined build depends

 -- Slamet Badwi (bedouin) <slamet.badwi@gmail.com>  Mon, 21 Apr 2014 23:20:50 +0700

alldeb-installer (0.1-ubuntu5) trusty; urgency=low

  * Add postinst and postrm script

 -- Slamet Badwi (bedouin) <slamet.badwi@gmail.com>  Mon, 21 Apr 2014 22:14:56 +0700

alldeb-installer (0.1-ubuntu4) trusty; urgency=low

  * Third attempt

 -- Slamet Badwi (bedouin) <slamet.badwi@gmail.com>  Mon, 21 Apr 2014 21:02:33 +0700

alldeb-installer (0.1-ubuntu3) trusty; urgency=low

  * Third packaging attempt

 -- Slamet Badwi (bedouin) <slamet.badwi@gmail.com>  Mon, 21 Apr 2014 20:49:34 +0700

alldeb-installer (0.1-ubuntu2) trusty; urgency=low

  * Initial release, second packaging attempt

 -- Slamet Badwi (bedouin) <slamet.badwi@gmail.com>  Mon, 21 Apr 2014 20:32:21 +0700

alldeb-installer (0.1-ubuntu1) trusty; urgency=low

  * Initial release

 -- Slamet Badwi (bedouin) <slamet.badwi@gmail.com>  Sun, 20 Apr 2014 02:35:16 +0700


alldeb_id.ts
<?xml version="1.0" encoding="utf-8"?>
<!DOCTYPE TS>
<TS version="2.1" language="id_ID">
<context>
    <name>About</name>
    <message>
        <location filename="about.ui" line="14"/>
        <source>Bantuan</source>
        <translation type="unfinished"></translation>
    </message>
    <message>
        <location filename="about.ui" line="72"/>
        <source>Tutup</source>
        <translation type="unfinished"></translation>
    </message>
    <message>
        <location filename="about.ui" line="79"/>
        <source>AllDebInstaller v0.2 (beta)</source>
        <translation type="unfinished"></translation>
    </message>
    <message>
        <location filename="about.ui" line="87"/>
        <source>Panduan</source>
        <translation type="unfinished"></translation>
    </message>
    <message>
        <location filename="about.ui" line="93"/>
        <source>&lt;!DOCTYPE HTML PUBLIC &quot;-//W3C//DTD HTML 4.0//EN&quot; &quot;http://www.w3.org/TR/REC-html40/strict.dtd&quot;&gt;
&lt;html&gt;&lt;head&gt;&lt;meta name=&quot;qrichtext&quot; content=&quot;1&quot; /&gt;&lt;style type=&quot;text/css&quot;&gt;
p, li { white-space: pre-wrap; }
&lt;/style&gt;&lt;/head&gt;&lt;body style=&quot; font-family:&apos;Sans Serif&apos;; font-size:9pt; font-weight:400; font-style:normal;&quot;&gt;
&lt;p style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;Program ini merupakan installer untuk file alldeb yang dibuat dengan alldeb_maker. Keterangan lengkap tentang alldeb ada di sini:&lt;/p&gt;
&lt;p style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;&lt;a href=&quot;http://ubuntu-indonesia.com/forums/ubbthreads.php/topics/129832/1&quot;&gt;&lt;span style=&quot; text-decoration: underline; color:#0000ff;&quot;&gt;http://ubuntu-indonesia.com/forums/ubbthreads.php/topics/129832/1&lt;/span&gt;&lt;/a&gt;&lt;br /&gt;&lt;/p&gt;
&lt;p style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;Dan alldeb_maker juga sudah dibuatkan PPA di: &lt;a href=&quot;https://launchpad.net/~faqihjakha/+archive/sikucinglewat&quot;&gt;&lt;span style=&quot; text-decoration: underline; color:#0000ff;&quot;&gt;https://launchpad.net/~faqihjakha/+archive/sikucinglewat&lt;/span&gt;&lt;/a&gt;&lt;/p&gt;
&lt;p style=&quot; margin-top:12px; margin-bottom:12px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;File alldeb bisa anda peroleh dari: &lt;a href=&quot;http://ubuntu-indonesia.com/forums/ubbthreads.php/topics/131713/1&quot;&gt;&lt;span style=&quot; text-decoration: underline; color:#0000ff;&quot;&gt;http://ubuntu-indonesia.com/forums/ubbthreads.php/topics/131713/1&lt;/span&gt;&lt;/a&gt;&lt;/p&gt;
&lt;p style=&quot; margin-top:12px; margin-bottom:12px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;Jika anda mempunyai file alldeb yang diperoleh dari tautan di atas, maka cukup klik kanan file tersebut lalu pilih &lt;span style=&quot; font-style:italic;&quot;&gt;Open With AllDeb Installer&lt;/span&gt;.&lt;/p&gt;
&lt;p style=&quot; margin-top:12px; margin-bottom:12px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;Lalu klik Next untuk melihat informasi file alldeb. Lalu klik Next lagi untuk lanjut. Terakhir, klik Instal untuk memulai instalasi. Nanti akan diminta kata sandi sebanyak dua kali.&lt;/p&gt;
&lt;p style=&quot; margin-top:12px; margin-bottom:12px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;Jika ada kesalahan, anda bisa melaporkan file yang dibuat oleh program ini kepada kami.&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</source>
        <translation type="unfinished"></translation>
    </message>
    <message>
        <location filename="about.ui" line="111"/>
        <source>Tentang Program</source>
        <translation type="unfinished"></translation>
    </message>
    <message>
        <location filename="about.ui" line="117"/>
        <source>&lt;!DOCTYPE HTML PUBLIC &quot;-//W3C//DTD HTML 4.0//EN&quot; &quot;http://www.w3.org/TR/REC-html40/strict.dtd&quot;&gt;
&lt;html&gt;&lt;head&gt;&lt;meta name=&quot;qrichtext&quot; content=&quot;1&quot; /&gt;&lt;style type=&quot;text/css&quot;&gt;
p, li { white-space: pre-wrap; }
&lt;/style&gt;&lt;/head&gt;&lt;body style=&quot; font-family:&apos;Sans Serif&apos;; font-size:9pt; font-weight:400; font-style:normal;&quot;&gt;
&lt;p style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;Program ini merupakan versi GUI untuk skrip alldeb_user. Versi GUI lainnya ditulis dengan toolkit FLTK:&lt;/p&gt;
&lt;p style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;&lt;a href=&quot;http://ubuntu-indonesia.com/forums/ubbthreads.php/topics/147894/1&quot;&gt;&lt;span style=&quot; text-decoration: underline; color:#0057ae;&quot;&gt;http://ubuntu-indonesia.com/forums/ubbthreads.php/topics/147894/1&lt;/span&gt;&lt;/a&gt;&lt;/p&gt;
&lt;p style=&quot;-qt-paragraph-type:empty; margin-top:12px; margin-bottom:12px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;&lt;br /&gt;&lt;/p&gt;
&lt;p style=&quot; margin-top:12px; margin-bottom:12px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;Lisensi: GPLv3&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</source>
        <translation type="unfinished"></translation>
    </message>
    <message>
        <location filename="about.ui" line="132"/>
        <source>Perancang</source>
        <translation type="unfinished"></translation>
    </message>
    <message>
        <location filename="about.ui" line="138"/>
        <source>&lt;!DOCTYPE HTML PUBLIC &quot;-//W3C//DTD HTML 4.0//EN&quot; &quot;http://www.w3.org/TR/REC-html40/strict.dtd&quot;&gt;
&lt;html&gt;&lt;head&gt;&lt;meta name=&quot;qrichtext&quot; content=&quot;1&quot; /&gt;&lt;style type=&quot;text/css&quot;&gt;
p, li { white-space: pre-wrap; }
&lt;/style&gt;&lt;/head&gt;&lt;body style=&quot; font-family:&apos;Sans Serif&apos;; font-size:9pt; font-weight:400; font-style:normal;&quot;&gt;
&lt;p style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;2013-2014&lt;/p&gt;
&lt;p style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;&lt;span style=&quot; font-weight:600;&quot;&gt;Nifa Dwi Kurniawan&lt;/span&gt;&lt;/p&gt;
&lt;p style=&quot; margin-top:0px; margin-bottom:10px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;(penulis skrip alldeb_maker dan alldeb_user serta versi FLTK)&lt;/p&gt;
&lt;p style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;2014&lt;/p&gt;
&lt;p style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;&lt;span style=&quot; font-weight:600;&quot;&gt;Slamet Badwi&lt;/span&gt;&lt;/p&gt;
&lt;p style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;(porting kode skrip alldeb_user ke Qt)&lt;/p&gt;
&lt;p style=&quot; margin-top:12px; margin-bottom:12px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;Laman Proyek: &lt;a href=&quot;https://launchpad.net/alldeb-installer&quot;&gt;&lt;span style=&quot; text-decoration: underline; color:#0057ae;&quot;&gt;https://launchpad.net/alldeb-installer&lt;/span&gt;&lt;/a&gt;&lt;/p&gt;
&lt;p style=&quot; margin-top:12px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;Tim Penerjemahan:&lt;/p&gt;
&lt;p style=&quot; margin-top:0px; margin-bottom:12px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;&lt;a href=&quot;https://www.transifex.com/projects/p/alldeb-installer/&quot;&gt;&lt;span style=&quot; text-decoration: underline; color:#0057ae;&quot;&gt;https://www.transifex.com/projects/p/alldeb-installer/&lt;/span&gt;&lt;/a&gt;&lt;/p&gt;
&lt;p style=&quot; margin-top:12px; margin-bottom:12px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;Bahasa Inggris: &lt;span style=&quot; font-weight:600;&quot;&gt;Ikhwan Setiawan&lt;/span&gt;&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</source>
        <translation type="unfinished"></translation>
    </message>
</context>
<context>
    <name>Dialog</name>
    <message>
        <location filename="dialog.ui" line="14"/>
        <source>AllDebInstaller</source>
        <translation>AllDebInstaller</translation>
    </message>
    <message>
        <location filename="dialog.ui" line="25"/>
        <source>Choose alldeb file</source>
        <translation>Pilih file alldeb</translation>
    </message>
    <message>
        <location filename="dialog.ui" line="32"/>
        <source>Select</source>
        <translation>Pilih</translation>
    </message>
    <message>
        <location filename="dialog.ui" line="112"/>
        <source>The contents of the alldeb file including dependencies. Please make sure that MD5 hash sum is correct.</source>
        <translation>Isi dari file allden termasuk dependensinya. Pastikan MD5 hash sum benar.</translation>
    </message>
    <message>
        <location filename="dialog.ui" line="132"/>
        <source>Total</source>
        <translation>Jumlah</translation>
    </message>
    <message>
        <location filename="dialog.ui" line="158"/>
        <source>Size</source>
        <translation>Ukuran</translation>
    </message>
    <message>
        <location filename="dialog.ui" line="184"/>
        <source>md5sum:</source>
        <translation>md5sum:</translation>
    </message>
    <message>
        <location filename="dialog.ui" line="208"/>
        <source>Click Install button to begin installing.</source>
        <translation>Klik tombol Instal untuk mulai menginstal.</translation>
    </message>
    <message>
        <location filename="dialog.ui" line="268"/>
        <source>Report</source>
        <translation>Laporkan</translation>
    </message>
    <message>
        <location filename="dialog.ui" line="275"/>
        <source>Quit</source>
        <translation>Keluar</translation>
    </message>
    <message>
        <location filename="dialog.ui" line="287"/>
        <source>Back</source>
        <translation>Mundur</translation>
    </message>
    <message>
        <location filename="dialog.ui" line="299"/>
        <location filename="dialog.cpp" line="161"/>
        <location filename="dialog.cpp" line="650"/>
        <source>Next</source>
        <translation>Maju</translation>
    </message>
    <message>
        <location filename="dialog.cpp" line="26"/>
        <source>About</source>
        <translation>Tentang</translation>
    </message>
    <message>
        <location filename="dialog.cpp" line="28"/>
        <source>Guide</source>
        <translation>Panduan</translation>
    </message>
    <message>
        <location filename="dialog.cpp" line="32"/>
        <source>Indonesian</source>
        <translation>Indonesian</translation>
    </message>
    <message>
        <location filename="dialog.cpp" line="35"/>
        <source>English</source>
        <translation>English</translation>
    </message>
    <message>
        <location filename="dialog.cpp" line="139"/>
        <location filename="dialog.cpp" line="150"/>
        <source>Select an alldeb file</source>
        <translation>Pilih satu file alldeb</translation>
    </message>
    <message>
        <location filename="dialog.cpp" line="139"/>
        <location filename="dialog.cpp" line="150"/>
        <source>AllDeb file (*.alldeb)</source>
        <translation>File Alldeb (*.alldeb)</translation>
    </message>
    <message>
        <location filename="dialog.cpp" line="199"/>
        <source>bytes</source>
        <translation>bytes</translation>
    </message>
    <message>
        <location filename="dialog.cpp" line="229"/>
        <source>An error occured</source>
        <translation>Terjadi kesalahan</translation>
    </message>
    <message>
        <location filename="dialog.cpp" line="304"/>
        <source>Error</source>
        <translation>Kesalahan</translation>
    </message>
    <message>
        <location filename="dialog.cpp" line="305"/>
        <source>This is not alldeb file we expected. Don&apos;t proceed, please!</source>
        <translation>Ini bukan file alldeb yang bisa diproses. Mohon jangan dilanjut!</translation>
    </message>
    <message>
        <location filename="dialog.cpp" line="318"/>
        <source>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;h2&gt;Installation&lt;/h2&gt;&lt;p&gt;You are going to install the following meta-package(s):&lt;/p&gt;&lt;p&gt;&lt;strong&gt;%1&lt;/strong&gt;&lt;/p&gt;&lt;p&gt;&lt;br/&gt;&lt;/p&gt;&lt;p&gt;Always make sure that this alldeb file is from a reliable source.&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</source>
        <translation>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;h2&gt;Installation&lt;/h2&gt;&lt;p&gt;Anda hendak menginstal meta-paket berikut ini:&lt;/p&gt;&lt;p&gt;&lt;strong&gt;%1&lt;/strong&gt;&lt;/p&gt;&lt;p&gt;&lt;br/&gt;&lt;/p&gt;&lt;p&gt;Selalu pastikan bahwa file ini dari sumber terpercaya.&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</translation>
    </message>
    <message>
        <location filename="dialog.cpp" line="386"/>
        <source>Cannot checking</source>
        <translation>Tidak bisa memeriksa</translation>
    </message>
    <message>
        <location filename="dialog.cpp" line="387"/>
        <source>Tanpa program pemindai paket, proses ini tidak bisa dilanjutkan.
Program yang dimaksud adalah apt-ftparchive (dari paket apt-utils) atau dpkg-scan-packages</source>
        <translation type="unfinished"></translation>
    </message>
    <message>
        <location filename="dialog.cpp" line="468"/>
        <source>Error occured</source>
        <translation>Terjadi kesalahan</translation>
    </message>
    <message>
        <location filename="dialog.cpp" line="498"/>
        <source>=================
Process failed
Please report this error by clicking the Report button below
=================
This report was created on %1</source>
        <translation>=================
Proses gagal
Mohon laporkan kesalahan ini dengan mengklik tombol Laporkan di bawah ini
=================
Laporan ini dibuat pada %1</translation>
    </message>
    <message>
        <location filename="dialog.cpp" line="511"/>
        <source>---------------------
Process was relocated to terminal emulator.</source>
        <translation>---------------------
Proses telah dipindahkan ke terminal emulator.</translation>
    </message>
    <message>
        <location filename="dialog.cpp" line="516"/>
        <source>---------------------
Process has been finished.</source>
        <translation>---------------------
Proses telah selesai.</translation>
    </message>
    <message>
        <location filename="dialog.cpp" line="525"/>
        <source>Extracting: </source>
        <translation>Mengekstrak: </translation>
    </message>
    <message>
        <location filename="dialog.cpp" line="574"/>
        <source>AllDebInstaller - Delete cache?</source>
        <translation>AllDebInstaller - Hapus tembolok?</translation>
    </message>
    <message>
        <source>Delete cache?</source>
        <translation type="obsolete">Hapus tembolok?</translation>
    </message>
    <message>
        <location filename="dialog.cpp" line="575"/>
        <source>Do you want to remove temporary files in:
</source>
        <translation>Apakah file tembolok ingin dihapus: </translation>
    </message>
    <message>
        <location filename="dialog.cpp" line="577"/>
        <source>Yes</source>
        <translation>Ya</translation>
    </message>
    <message>
        <location filename="dialog.cpp" line="578"/>
        <source>No</source>
        <translation>Tidak</translation>
    </message>
    <message>
        <location filename="dialog.cpp" line="610"/>
        <location filename="dialog.cpp" line="648"/>
        <source>Install</source>
        <translation>Instal</translation>
    </message>
    <message>
        <location filename="dialog.cpp" line="670"/>
        <source>A report file had been created in:
%1
Entitled alldeb-report.txt

Please send it to author&apos;s email or to project&apos;s Bug reporting homepage in Launchpad.
We&apos;re sorry for this inconvenience.</source>
        <translation>File laporan telah dibuat di:
%1
Dengan nama alldeb-report.txt

Silakan kirimkan ke surel penulis atau laman pelaporan Bug di Launchpad.
Mohon maaf atas ketidaknyamanan ini.</translation>
    </message>
</context>
</TS>
